app/api/graphql/route.ts

```diff
@@ -0,0 +1,100 @@
+import { connectToMongo } from "@/lib/db/mongo";
+import { connectToPostgres } from "@/lib/db/postgres";
+import {
+  authorQueryResolver,
+  createAuthorMutationResolver,
+  deleteAuthorMutationResolver,
+  updateAuthorMutationResolver,
+} from "@/lib/resolvers/author";
+import {
+  bookQueryResolver,
+  createBookMutationResolver,
+  deleteBookMutationResolver,
+  updateBookMutationResolver,
+} from "@/lib/resolvers/book";
+import {
+  createReviewMutationResolver,
+  reviewQueryResolver,
+} from "@/lib/resolvers/review";
+import {
+  authorMutation,
+  authorQuery,
+  authorType,
+  paginatedAuthorType,
+} from "@/lib/typeDefs/author";
+import {
+  bookMutation,
+  bookQuery,
+  bookType,
+  paginatedBookType,
+} from "@/lib/typeDefs/book";
+import {
+  reviewMutation,
+  reviewQuery,
+  reviewResponseType,
+  reviewType,
+} from "@/lib/typeDefs/review";
+import { once } from "@/lib/util/once";
+import { ApolloServer } from "@apollo/server";
+import { startServerAndCreateNextHandler } from "@as-integrations/next";
+
+const typeDefs = `
```

Comment:
any specific reason for declaring typedef like this ? instead can't we directly do 
`const typeDefs = fs.readFileSync(
    path.join(path.resolve(), "./schema.graphql"),
    "utf8"
);`

Reply:
No specific reason. I was following a tutorial but I realise having a schema file is much cleaner.

Reply:
One other way to extend this, is how we do in sprinto. We basically say that all types, queries resolves need to be written in `.gql` files, and then we fs read through all the files combining them. So if you create a new file, and just write contextual types, it will get added to the global schema